### ISSUE TITLE
No rom showing : Darknior

### ISSUE THREAD

**Issue by Darknior** [lashower/RetroPieWebConsole#1]
```
You have triggered an unhandledRejection, you may have forgotten to catch a Promise rejection:
Error: ENOENT: no such file or directory, scandir '/home/root/RetroPie/roms'
```

I  not understand ?

[RetroPieWebConsole-error-0.log](https://github.com/lashower/RetroPieWebConsole/files/1550838/RetroPieWebConsole-error-0.log)
[RetroPieWebConsole-out-0.log](https://github.com/lashower/RetroPieWebConsole/files/1550839/RetroPieWebConsole-out-0.log)

**Comment by lashower** [comment]
I see your problem. So RetroPieWebConsole utilizes process.env.SUDO_USER, just like RetroPie-Setup. Basically, you are running RPWC as root instead of in sudo as pi, which is what I expected.

Does your home page show User as root or pi like mine?
![image](https://user-images.githubusercontent.com/11881208/33887978-98e0804c-df19-11e7-81d8-5150139d0b31.png)

**Comment by lashower** [comment]
I added a patch to switch SUDO_USER to pi if set to root.

Go ahead and do an update and see if it is working.

**Comment by Darknior** [comment]
Yes i'm with root user, i don't choose it, i don't know why it is root user.

**Comment by Darknior** [comment]
I have git pull but i"m always in root user :( Nothing works better.

**Comment by lashower** [comment]
That's fine. I added an override to switch user to pi if found as root.

Does the home page show pi now?

If so, then run a search for roms.

**Comment by Darknior** [comment]
Like i write you i have GIT PULL to update and i always read - User : root

**Comment by lashower** [comment]
Then try running this command. 
pm2 restart RetroPieWebConsole

Tell me if that worked. If not, run "less app.js | grep SUDO" and give me the output.

**Comment by Darknior** [comment]
`pm2 restart RetroPieWebConsole`

Restart the PM2 but i'm always log as **root**.

`
[PM2] Spawning PM2 daemon with pm2_home=/home/pi/.pm2
[PM2] PM2 Successfully daemonized
Use --update-env to update environment variables
[PM2][ERROR] Process RetroPieWebConsole not found
┌──────────┬────┬──────┬─────┬────────┬─────────┬────────┬─────┬─────┬──────┬──────────┐
│ App name │ id │ mode │ pid │ status │ restart │ uptime │ cpu │ mem │ user │ watching │
└──────────┴────┴──────┴─────┴────────┴─────────┴────────┴─────┴─────┴──────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
`
I launch : `less app.js | grep SUDO`

`if(process.env.SUDO_USER == 'root')
    process.env.SUDO_USER = 'pi';
            fs.chown(file.path,Number(process.env.SUDO_UID),Number(process.env.SUDO_UID)).then(() => {
`

**Comment by lashower** [comment]
That restart didn't work based on this:
[PM2][ERROR] Process RetroPieWebConsole not found 

Can you try this and post the output?
sudo pm2 restart RetroPieWebConsole
ps -ef | grep [R]etroPieWebConsole
EX:
```
pi@lucaspi:~/nodejs/RetroPieWebConsole` $ sudo pm2 restart RetroPieWebConsole
Use --update-env to update environment variables
[PM2] Applying action restartProcessId on app [RetroPieWebConsole](ids: 0)
[PM2] [RetroPieWebConsole](0) ✓
┌────────────────────┬────┬──────┬───────┬────────┬─────────┬────────┬─────┬───────────┬──────┬──────────┐
│ App name           │ id │ mode │ pid   │ status │ restart │ uptime │ cpu │ mem       │ user │ watching │
├────────────────────┼────┼──────┼───────┼────────┼─────────┼────────┼─────┼───────────┼──────┼──────────┤
│ RetroPieWebConsole │ 0  │ fork │ 12934 │ online │ 245     │ 0s     │ 66% │ 13.6 MB   │ root │ disabled │
└────────────────────┴────┴──────┴───────┴────────┴─────────┴────────┴─────┴───────────┴──────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
pi@lucaspi:~/nodejs/RetroPieWebConsole $ ps -ef | grep [R]etroPieWebConsole
root     12934   928 86 09:10 ?        00:00:02 node /home/pi/nodejs/RetroPieWebConsole/app.js
```

**Comment by Darknior** [comment]
sudo pm2 restart RetroPieWebConsole
```
Use --update-env to update environment variables
[PM2] Applying action restartProcessId on app [RetroPieWebConsole](ids: 0)
[PM2] [RetroPieWebConsole](0) ✓
┌────────────────────┬────┬──────┬───────┬────────┬─────────┬────────┬─────┬───────────┬──────┬──────────┐
│ App name           │ id │ mode │ pid   │ status │ restart │ uptime │ cpu │ mem       │ user │ watching │
├────────────────────┼────┼──────┼───────┼────────┼─────────┼────────┼─────┼───────────┼──────┼──────────┤
│ RetroPieWebConsole │ 0  │ fork │ 12038 │ online │ 2       │ 0s     │ 99% │ 14.5 MB   │ root │ disabled │
└────────────────────┴────┴──────┴───────┴────────┴─────────┴────────┴─────┴───────────┴──────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app

pi@retropie:~/src/Webtropie $ ps -ef | grep [R]etroPieWebConsole
root     12038  1223  8 15:58 ?        00:00:05 node /root/nodejs/RetroPieWebConsole/app.js
```

**Comment by lashower** [comment]
Now does the home page still show as user root?
![image](https://user-images.githubusercontent.com/11881208/33947077-7c699b0a-dff1-11e7-855c-dc30cf4d1fff.png)

**Comment by Darknior** [comment]
No, i will write it to you if it works.
Always ROOT.

**Comment by lashower** [comment]
Looks like I will have to create a preferences page for overwrites.

**Comment by lashower** [comment]
Finally got the settings page in.
FYI, the validation of folders is a little clunky. If any show as red, you will have to refresh the page after saving to validate if they exist or not.

![image](https://user-images.githubusercontent.com/11881208/34222835-0265cf82-e58b-11e7-98a8-a985fbf14fca.png)

**Comment by lashower** [comment]
Any update?

**Comment by Darknior** [comment]
Hi
Sorry for the delay but my PI totally crash :(
I have began to install again all my stuff but i must finish it.

And after i will clone the SD to an other to tests your **WebConsole**.
I don't want to add again to my system NodeJS n Mongo before verifying how many memory and CPU they use. They are always running on TOP ...
But yes i will finish to help you.

**Comment by lashower** [comment]
No worries. I appreciate all your help.
I will be looking into resource use as well.
I have been collecting some metrics. I think there is also some cleanup I can do to decrease the memory and CPU footprint.
